stack-graphs/src/cycles.rs

```diff
@@ -127,3 +131,103 @@ where
         true
     }
 }
+
+#[derive(Clone)]
+pub struct AppendingCycleDetector {
+    nodes: VecDeque<Handle<Node>>,
+}
+
+impl AppendingCycleDetector {
+    pub fn from_node(node: Handle<Node>) -> Self {
+        Self {
+            nodes: vec![node].into(),
+        }
+    }
+
+    pub fn appended(
+        &mut self,
+        graph: &StackGraph,
+        partials: &mut PartialPaths,
+        node: Handle<Node>,
+        new_path: &PartialPath,
+    ) -> bool {
+        if let Some(i) = self.nodes.iter().position(|n| *n == node) {
```

Comment:
One of my small pet peeves about Rust is how `if let` is not compatible with an early return style.  I personally find early-return more important, so I'd suggest changing this to:

```suggestion
        let i = match self.nodes.iter().position(|n| *n == node) {
            Some(i) => i,
            None => return true,
        };
```

And then you can dedent the body of the `if` statement

Reply:
Ah yes, it's a shame there's no neat construct for that. I agree that inverting it here is better.